### ISSUE TITLE
Hide some controls when not actionable

### ISSUE THREAD

**Issue by almet** [Kinto/formbuilder#181]
We should not show the save and reset buttons when no fields have been added yet, as they are un-effective. I'm not sure what the best way to deal with them: should they be greyed or now be shown at all? cc @natouille @QuentinRoy

**Comment by QuentinRoy** [comment]
I would recommend greying rather than hiding.
(1) Even if the action is currently not available, users can still discover that it is here and may become available depending on the system state (the grey feedback is widespread). It is even better if a feedback is provided to tell why the action is currently not possible (e.g. with a tooltip). 
(2) Hiding and showing elements like this on an application may be confusing. It is likely that a user won't "scan" the page a second time to see if it has changed once she created a question. I feel like it may even be more important for buttons like "save" as more and more applications / websites do not require one anymore. It is not impossible that some users fully miss it and make mistakes.
My subjective opinion on removing elements is that in general, it is OK if the user will never be exposed to them. The goal is to strip the interface down rather than forbidding the access to functionalities.

**Comment by almet** [comment]
Makes sense, thanks!

**Comment by Natouille** [comment]
Do you mean the first "page" below ?
![formbuilder_-_2016-10-20_11 46 45](https://cloud.githubusercontent.com/assets/2006947/19554940/f095550e-96ba-11e6-902a-cd0946a59e87.png)

If so, I don't see hiding the buttons as confusing. The page changes completely when you add a new element (short text, long text, etc.) 
![formbuilder_-_2016-10-20_11 49 37](https://cloud.githubusercontent.com/assets/2006947/19555008/50e3f910-96bb-11e6-8cdb-974b6162bcb0.png)

Most of the time I try to avoid these unclickable buttons, because it can be seen as a false promise ("it's there but I cannot do anything on it : why ?"). So I hide and then show them when the access is granted.

**Comment by QuentinRoy** [comment]
I agree, that in this case, it is almost as if one ends up in a fully new page so it probably does not matter much and you may want to make your first page cleaner.
However, I don't think the "it's there but I cannot do anything on it : why ?" is a big issue though. This is what [Tesler would call a mode problem](https://archive.org/details/TheSmalltalkEnvironment). But, hiding it does not help a user finding how to make an action. I agree it may avoid some frustration, but IMHO, that's only because the user does not know the action can be performed in the first place which, I think, is a discoverability issue way worse than this frustration. But then hey, that's HCI, no hard rules here.

**Comment by almet** [comment]
Please note that the currently deployed version on fourmilieres is not what the UI currently looks like. I've yet to deploy the new version there, but you can see it at http://kinto.github.io/formbuilder

I think it solves a few problems but I feel the previous version was easier to grasp (we've added a click more to add a field!). I would love to have some feedback.

**Comment by QuentinRoy** [comment]
Are you sure you do not want to try a bit of paper prototyping guys? You know just roughly draw how the interface could look like on paper and trying out different stuffs before jumping into the code. I feel like you are doing exactly what we are telling our students not to do :).

**Comment by QuentinRoy** [comment]
I personally tend to prefer the interface as it was before, but I do think you guys could use a bit more prototyping, iteration, user testing (even quick or informal) on this. No "UX expert" can replace that.

**Comment by almet** [comment]
I think we'll conduct user testing research in the future, but we first need to get to a point where we have something to show them :-)

**Comment by DirtyF** [comment]
@almet you can show them paper prototype, these can be added in apps to test user scenarios, there are a lot of ways to engage users before writing any code.

 @QuentinRoy I'd be interested in conducting some user-testing based on lo-fi prototypes, maybe it's something we could do without interfering with the current development? What do you think cc/ @Natouille

**Comment by almet** [comment]
We'll try to organise something here in rennes with @rlecellier, and share our results. I'm interested in anything you find on your side, that would be great!

The biggest challenge for us here is that we never did this, so we're scared ;)

**Comment by QuentinRoy** [comment]
That is very very easy. If you want, let's Skype this WE I can tell you how to get started. In the meantime, and for those who are not convinced yet or think that won't fit well in your timing, [you can have a look at this](https://www.nngroup.com/articles/mozilla-paper-prototype/). Funnily, it looks quite appropriate.

**Comment by DirtyF** [comment]
@almet great to hear, we could run a session here in Toulouse too with @Natouille. but I'd like to have @QuentinRoy input on prototypes.

**Comment by Natouille** [comment]
@almet don't be scared :) It should be quite easy to draw some prototypes in order to test them with users.

**Comment by rlecellier** [comment]
What about something like this ?

![untitled_page](https://cloud.githubusercontent.com/assets/139848/19560624/852a4df4-96d5-11e6-8a2b-341659b6c194.png)

**Comment by Natouille** [comment]
@rlecellier I think it's a great idea to use the standard web text fields. I am absolutely not a fan of the new Material Design look of these fields. If I check on this page below, I don't what's editable and what's not 
![formbuilder_-_2016-10-20_17 40 08](https://cloud.githubusercontent.com/assets/2006947/19566921/4eaa597e-96ec-11e6-80c6-4bde924972fb.png)
For you other options, I have to check, there are some good ideas here. 
To all : it would be better if we could agree on a global reflection rather than just some tiny adjustments here and there. At the end, it will be a bit messy.

**Comment by rlecellier** [comment]
Here my last proposal with the two current version. You can either : 
* propose new stuff then i'll add a build the layout with pencil.
* propose modifications on one of the three layouts then i'll modify it.

Currently on https://www.fourmilieres.net/#/
![add_field_001](https://cloud.githubusercontent.com/assets/139848/19568192/e36c040a-96f0-11e6-8e74-22b52c0f4453.png)

Currently on http://kinto.github.io/formbuilder/#/builder
![add_field_002](https://cloud.githubusercontent.com/assets/139848/19568201/e912d668-96f0-11e6-88c8-2e1e53b2bcee.png)
